tests/testthat/test-lint_package.R

```diff
@@ -91,7 +91,11 @@ test_that(
     )
     lints_from_a_subdir <- withr::with_dir(
       file.path(pkg_path, "R"),
-      lint_package("..", linters = list(assignment_linter()))
+      lint_package(".", linters = list(assignment_linter()))
```

Comment:
This now tests two things: That exclusions are relative to the `.lintr` config and that `lint_package()` finds the package from within as intended.